src/Raven.Client/Server/DatabaseTopology.cs

```diff
@@ -45,15 +117,36 @@ public DynamicJsonValue ToJson()
             {
                 [nameof(Members)] = new DynamicJsonArray(Members),
                 [nameof(Promotables)] = new DynamicJsonArray(Promotables),
-                [nameof(Watchers)] = new DynamicJsonArray(Watchers)
+                [nameof(Watchers)] = new DynamicJsonArray(Watchers.Select(w => w.ToJson())),
+                [nameof(CustomConnectionBlocker)] = DynamicJsonValue.Convert(CustomConnectionBlocker),
+                [nameof(NameToUrlMap)] = DynamicJsonValue.Convert(NameToUrlMap)
             };
         }
 
         public void RemoveFromTopology(string delDbFromNode)
         {
             Members.Remove(delDbFromNode);
             Promotables.Remove(delDbFromNode);
-            Watchers.Remove(delDbFromNode);
+            CustomConnectionBlocker.Remove(delDbFromNode);
+        }
+
+        public bool IsItMyTask(IDatabaseTask task, string nodeTag)
+        {
+            var myPosition = Members.FindIndex(s => s == nodeTag);
```

Comment:
If you are not a member you can't be assigned with task.
So watchers/promotables (and suspects) are passive "players" in the topology.